src/commands/path/path.generate.ts

```diff
@@ -0,0 +1,215 @@
+  name: 'generate-path',
+  description: 'Generate target path from collection metadata',
+  version: CliInfo.version,
+  args: {
+    config,
+    verbose,
+
+    targetBucketName: option({
+      type: string,
+      long: 'target-bucket-name',
+      description: 'Target bucket name, e.g. nz-imagery',
+    }),
+
+    source: positional({
+      type: string,
+      displayName: 'path',
+      description: 'path to source data where collection.json file is located',
+    }),
+  },
+
+  async handler(args) {
+    registerCli(this, args);
+    const startTime = performance.now();
+
+    if (args.targetBucketName.startsWith('s3://')) {
+      throw new Error(
+        `target-bucket-name entered as path, did you mean ${args.targetBucketName
+          .replace('s3://', '')
+          .replace('/', '')}?`,
+      );
+    }
+
+    logger.info({ source: args.source }, 'GeneratePath:Start');
+
+    const collection = await fsa.readJson<StacCollection>(fsa.join(args.source, 'collection.json'));
+    if (collection == null) throw new Error(`Failed to get collection.json from ${args.source}.`);
+
+    const tiff = await loadFirstTiff(args.source, collection);
+
+    const category = getCategory(collection);
+    const geographicDescription = getGeographicDescription(collection);
+    const region = getRegion(collection);
+    const event = getEvent(collection);
+    const date = getDate(collection);
+    const gsd = extractGsd(tiff);
+    const epsg = extractEpsg(tiff);
+
+    const target = generatePath(args.targetBucketName, category, geographicDescription, region, event, date, gsd, epsg);
+    logger.info({ duration: performance.now() - startTime, target: target }, 'GeneratePath:Done');
+    return target;
+  },
+});
+
+/**
+ * Generates Target Path based on category
+ *
+ * @param {string} targetBucketName
+ * @param {string} category
+ * @param {string} geospaital_description
+ * @param {string} region
+ * @param {string} event
+ * @param {string} date
+ * @param {string} gsd
+ * @param {string} epsg
+ * @returns {string}
+ */
+export function generatePath(
+  targetBucketName: string,
+  category: string,
+  geospaital_description: string,
+  region: string,
+  event: string,
+  date: string,
+  gsd: string,
+  epsg: string,
+): string {
+  const name = generateName(region, geospaital_description, event);
+
+  if (category === dataCategories.SCANNED_AERIAL_PHOTOS) {
+    // nb: Historic Imagery is out of scope as survey number is not yet recorded in collection metadata (15/02/24)
+    throw new Error(`Automated target generation not implemented for historic imagery`);
+  } else if ([dataCategories.URBAN_AERIAL_PHOTOS, dataCategories.RURAL_AERIAL_PHOTOS].includes(category)) {
+    return `s3://${targetBucketName}/${region}/${name}_${date}_${gsd}/rgb/${epsg}/`;
+  } else if (category === dataCategories.SATELLITE_IMAGERY) {
+    return `s3://${targetBucketName}/${region}/${name}_${date}_${gsd}/rgb/${epsg}/`;
```

Comment:
Satellite imagery is currently following the same naming convention as other imagery however this will no be consistent with what is already on the ODR 
What currently exists:
```
s3://nz-imagery/new-zealand/north-island_20221122_10m/rgb/2193/
s3://nz-imagery/new-zealand/north-island_20230220_10m/rgb/2193/
s3://nz-imagery/new-zealand/north-island_2023_0.5m/rgb/2193/
```
what this might look like with new conventions:
```
s3://nz-imagery/new-zealand/north-island-cylcone-gabrielle_2022_10m/rgb/2193/
s3://nz-imagery/new-zealand/north-island-cyclone-gabirelle_2023_10m/rgb/2193/
s3://nz-imagery/new-zealand/north-island-cyclone-gabirelle_2023_0.5m/rgb/2193/
```
This change might require updates for https://github.com/linz/imagery/blob/master/docs/naming.md

Reply:
imagery target paths seem to match this convention for example:
`s3://nz-imagery/nelson/top-of-the-south-flood_2022_0.15m/rgb/2193/` 
region: `nelson` geographicDescription: `top-of-the-south` event: Flood`

`s3://nz-imagery/hawkes-bay/hawkes-bay-cyclone-gabrielle_2023_0.1m/rgb/2193/`
region: `hawkes-bay` geographicDescription: `` event: `Cyclone Gabrielle`

`s3://nz-imagery/gisborne/gisborne-cyclone-gabrielle_2023_0.2m/rgb/2193/`
region: `gisborne` geographicDescription: `` event: `Cyclone Gabrielle`

`s3://nz-imagery/canterbury/christchurch-earthquake_2011_0.1m/rgb/2193/`
region: `canterbury` geographicDescription: `Christchurch` event: `Earthquake`

but not:

`s3://nz-imagery/auckland/auckland-coast_2022-2023_0.05m/rgb/2193` - is this flood imagery?
region: `auckland` geographicDescription: `Auckland Coast` event: ``

There is no mention of event in the [naming doc](https://github.com/linz/imagery/blob/master/docs/naming.md)?

Reply:
This structure will change to:
event, if no event geographic desc, if no geographic desc then region however I will make this change in a separate PR